File: js/insertionsort.js
var insertionsort = function (arr) {
	for (var i = 1; i < arr.length; i++) {
		var x = arr[i];
		j = i;

		while (j > 0 && arr[j - 1] > x) {
			arr[j] = arr[j - 1];
			j--;
		}

		arr[j] = x;
	}

	return arr;
};

var array = [6, 4, 2, 3, 5, 2, 10, 9];
console.log(insertionsort(array));

File: c/insertionsort.c
#include <stdio.h>
#include <stdlib.h>

void insertionsort(int a[], int n);

int main()
{
	int i;
	int a[10] = {5, 6, 3, 2, 8, 10, 4, 1, 9, 7};

	for (i = 0; i < 10; i++)
		printf("%d\t", a[i]);
	printf("\n");

	insertionsort(a, 10);

	for (i = 0; i < 10; i++)
		printf("%d\t", a[i]);
	printf("\n");

	return 0;
}

void insertionsort(int a[], int n)
{
	int i, j, p, x;

	for (i = 1; i < n; i++)
	{
		x = a[i];
		j = i;

		while (j > 0 && a[j - 1] > x)
		{
			a[j] = a[j - 1];
			j--;
		}
			
		a[j] = x;
	}
}
